apps/remote_control/lib/lexical/remote_control/project_node.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Lexical.RemoteControl.ProjectNode do
+  alias Lexical.RemoteControl
+  require Logger
+
+  defmodule State do
+    defstruct [:project, :paths, :cookie]
+
+    def new(project) do
+      paths = format_prepending_paths(RemoteControl.glob_paths())
+      cookie = Node.get_cookie()
+
+      struct!(__MODULE__, paths: paths, cookie: cookie, project: project)
+    end
+
+    defp format_prepending_paths(paths_as_charlists) do
+      Enum.map_join(paths_as_charlists, " -pa ", &Path.expand/1)
+    end
+  end
+
+  alias Lexical.RemoteControl.ProjectNodeSupervisor
+
+  use GenServer
+
+  def wait_until_started(project, project_listener, boot_timeout \\ 5_000) do
+    :ok = :net_kernel.monitor_nodes(true, node_type: :visible)
+
+    {:ok, node_pid} =
+      DynamicSupervisor.start_child(
+        ProjectNodeSupervisor,
+        {RemoteControl.ProjectNode, project}
+      )
+
+    node = RemoteControl.node_name(project)
+    remote_control_config = Application.get_all_env(:remote_control)
+
+    with :ok <- wait_until(boot_timeout),
+         :ok <-
+           :rpc.call(node, RemoteControl.Bootstrap, :init, [
+             project,
+             project_listener,
+             remote_control_config
+           ]) do
+      {:ok, node_pid}
+    end
```

Comment:
this should all be done inside the genserver to ensure requests are serialized. What would happen if this is called twice in rapid succession?

Additionally, all the waits would be eliminated in favor of responding to `nodeup` messages and doing a delayed reply with `GenServer.reply`